src/wallet/CarbonWallet.ts

```diff
@@ -324,6 +329,15 @@ export class CarbonWallet {
     return wallet;
   }
 
+  public static withRainbowKit(rainbowKit: RainbowKitAccount, evmChainId: string, compressedPubKeyBase64: string, addressOptions: SWTHAddressOptions, walletProvider: string, opts: Omit<CarbonWalletInitOpts, "signer"> = {}) {
```

Comment:
For `walletProvider`, think it would be good to use enum `SupportedEip6963Provider` instead of string to prevent users from entering a random string.

i.e.
```typescript
export interface RainbowKitWalletOpts {
  ...
  walletProvider: SupportedEip6963Provider;
}
```